src/cli/main.rs

```diff
@@ -198,6 +204,15 @@ fn format_string(
     }
 }
 
+fn get_ignore(directory: &Path) -> Result<Gitignore> {
+    let file_path = find_ignore_file_path(directory.to_path_buf(), true).unwrap();
+    if let Some(file_path) = file_path {
+        Ok(Gitignore::new(file_path).0)
```

Comment:
The second return to Gitignore::new is typically an error. Instead of ignoring we should surface it up to the user appropriately

Reply:
What is the standard pattern for surfacing? Log it to stderr? Panic?

Reply:
We use anyhow for our errors, which allows us to propagate it upwards and report it properly.

I think gitignore returns a `Vec<Error>` which might need slightly tweaking. You can then `return Err(...)` with the useful info

Check out some other errors, we typically stick `.context` or `.with_context` on the error to provide better info contextual info.

This might be useful: https://docs.rs/anyhow/latest/anyhow/

Reply:
Nevermind, it's an Option<Error>, which is slightly nicer.

You should just be able to check if error exists and then `return Err(error)`. Potentially with some extra context

Reply:
Another question. The errors return from `Gitignore::new` are actually partial errors. Say there's one invalid line in the ignore file, that error will be returned as the second parameter. Should we abort the whole operation because of that partial error? Or just keep going?

Reply:
Personally I think it's better to just abort completely, because even if there is a partial error that error could be important for the particular file we are formatting

Reply:
Updated. It will now abort the process with `("failed to parse ignore file: {:#}", err`) if error is returned from `Gitignore::new`.